### DISCUSSION TITLE
Changing PhotoSwipe's Default UI Icons

### DISCUSSION THREAD

**Discussion by anipalur** [CaiJimmy/hugo-theme-stack#871] [Q&A]
I would like to change the default PhotoSwipe UI icons (zoom, maximise, close, etc.) when viewing images to use [Tabler Icons](https://tabler-icons.io "Learn more about Tabler Icons.") instead. As of now, the icons don't match the rest of the site.

![Default PhotoSwipe UI icons](https://github.com/CaiJimmy/hugo-theme-stack/assets/135231141/6027b85c-9886-4c60-bec5-dc9c1d8b9aed)

I understand that PhotoSwipe JS and CSS are sourced from a CDN here:

https://github.com/CaiJimmy/hugo-theme-stack/blob/ce0a252bfa76a1e6f53d3b8a830e9cf61ee0c977/data/external.yaml#L6-L21

However, I couldn't find a way to change the icons from here. I read [PhotoSwipe's documentation](https://photoswipe.com/styling "View PhotoSwipe's documentation.") on this too but didn't know where to add the JS. Any help would be appreciated.

P.S. It would be great if PhotoSwipe UI icons were sourced from [assets/icons](https://github.com/CaiJimmy/hugo-theme-stack/tree/master/assets/icons "View the icons directory.") like other icons so users could easily change them.

**Comment by Lovirr** [comment]
here is the source code of PhotoSwipe:[https://github.com/dimsemenov/PhotoSwipe/archive/refs/tags/v4.1.3.zip](https://github.com/dimsemenov/PhotoSwipe/archive/refs/tags/v4.1.3.zip)

we can define our icons by downloading the files from the CDN and modifying them. Upon my examination, the original SVG icons are not separate; instead, four icons are combined into one image, and a specific icon is displayed by shifting it left or right. The exact location is at:

[https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.svg](https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.svg)

![image](https://github.com/CaiJimmy/hugo-theme-stack/assets/106920260/1bc34926-270f-4ffe-b62d-adca92572513)

However, it appears white and is not visible against a white background. If you want to preview it, please visit:
[https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.png](https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.png)

If you wish to modify the icons, you need to search for the URL in the default-skin.min.css file:
https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.min.css

![image](https://github.com/CaiJimmy/hugo-theme-stack/assets/106920260/ef143bad-0f2a-4c13-bcbd-9d7d9b49248c)


Then, choose your favourite icon to replace the origin one, and adjust the spacing between the icons to display the desired one.

Overall, this is a rather cumbersome method.
